client/test/router_store.spec.js

```diff
@@ -0,0 +1,669 @@
+      { id: 'mb4', label: 'draft', attribs: '', tree: [''], accountID: 'a1' },
+    ],
+  },
+  lastPage: {
+    info: 'last-page',
+    isComplete: true,
+  },
+  modal: {
+    display: true,
+  },
+  message1: {
+    id: 'i1',
+    accountID: 'a1',
+    messageID: 'me1',
+    flags: [MessageFlags.SEEN],
+    conversationID: 'c1',
+    mailboxIDs: { mb1: 1 },
+  },
+  message2: {
+    id: 'i2',
+    accountID: 'a1',
+    messageID: 'me2',
+    flags: [MessageFlags.SEEN],
+    conversationID: 'c2',
+    mailboxIDs: { mb1: 1 },
+  },
+  message3: {
+    id: 'i3',
+    accountID: 'a1',
+    messageID: 'me3',
+    conversationID: 'c3',
+    mailboxIDs: { mb1: 1, mb3: 1 },
+    flags: [MessageFlags.FLAGGED, MessageFlags.ATTACH],
+  },
+  message4: {
+    id: 'i4',
+    accountID: 'a1',
+    messageID: 'me4',
+    conversationID: 'c4',
+    mailboxIDs: { mb4: 1 },
+    flags: [MessageFlags.FLAGGED, MessageFlags.ATTACH],
+  },
+
+  router: {
+    navigate: (url) => {
+      currentURL = url;
+    },
+    routes: {
+      'mailbox/:mailboxID(?:query)': 'messageList',
+      'account/new': 'accountNew',
+      'account/:accountID/settings/:tab': 'accountEdit',
+      'mailbox/:mailboxID/new': 'messageNew',
+      'mailbox/:mailboxID/:messageID/edit': 'messageEdit',
+      'mailbox/:mailboxID/:messageID/forward': 'messageForward',
+      'mailbox/:mailboxID/:messageID/reply': 'messageReply',
+      'mailbox/:mailboxID/:messageID/reply-all': 'messageReplyAll',
+      'mailbox/:mailboxID/:conversationID/:messageID(?:query)': 'messageShow',
+      '': 'defaultView',
+    },
+  },
+};
+
+
+describe('Router Store', () => {
+  let routerStore;
+  let dispatcher;
+
+
+  function changeRoute(query, message) {
+    if (message === undefined) message = fixtures.message1;
+    dispatcher.dispatch({
+      type: ActionTypes.ROUTE_CHANGE,
+      value: {
+        accountID: fixtures.account.id,
+        mailboxID: fixtures.account.inboxMailbox,
+        tab: 'selected',
+        action: MessageActions.SHOW,
+        conversationID: message.conversationID,
+        messageID: message.id,
+        query,
+      },
+    });
+  }
+
+  function loadMessages(messages, conversationLength) {
+    dispatcher.dispatch({
+      type: ActionTypes.MESSAGE_FETCH_SUCCESS,
+      value: {
+        lastPage: fixtures.lastPage,
+        result: {
+          messages,
+          conversationLength,
+        },
+        timestamp: new Date(),
+      },
+    });
+  }
+
+  function generateMessages() {
+    const msgs = [];
+    for (let i = 0; i < UtilConstants.MSGBYPAGE + 3; i++) {
+      const message = _.clone(fixtures.message1);
+      message.id = `id${i}`;
+      message.messageID = `meid${i}`;
+      message.conversationID = `c${i}`;
+      msgs.push(message);
+    }
+    return msgs;
+  }
+
+  /*
+   * Problem noticed:
+   *
+   * * FIXME: Private methods are melted with public ones.
+   * * FIXME: Store requests other stores without using getters.
```

Comment:
`Getters` are not for `stores` but for `components`. The rules to call stores is: only `RouterStore` and `actionCreators` can call stores. It is usefull to create the context of server request or know default status values. Other `stores`(that are fake ones) can't communicate with other stores.

SO you can remove this comment :)